softwareComponents/rofiHalSimPy/hal.cpp

```diff
@@ -0,0 +1,90 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/functional.h>
+
+#include <rofi_hal.hpp>
+
+namespace py = pybind11;
+
+using namespace rofi::hal;
+
+PYBIND11_MODULE( pyRofiHal, m ) {
+    m.doc() = "RoFI simulator HAL";
+
+    py::class_< Joint > joint( m, "Joint" );
+    joint
+        .def_property_readonly( "maxPosition", &Joint::maxPosition )
+        .def_property_readonly( "minPosition", &Joint::minPosition )
+        .def_property_readonly( "maxSpeed", &Joint::maxSpeed )
+        .def_property_readonly( "minSpeed", &Joint::minSpeed )
+        .def_property_readonly( "maxTorque", &Joint::maxTorque )
+        .def_property( "velocity", &Joint::getVelocity, &Joint::setVelocity )
+        .def_property( "torque", &Joint::getTorque, &Joint::setTorque )
```

Comment:
I would split `setVelocity` and `setTorque` - as it sets the movement type - so it effects other values as well (and to keep it consistent with `setPosition`).

Reply:
Makes sense as setting the values also affects the mode the joints are working in.